dev/tools/svg2dart/lib/svg2dart.dart

```diff
@@ -0,0 +1,547 @@
+    }
+    return new SvgPath(id, commands);
+  }
+
+  SvgPath applyTransform(_Transform transform) {
+    final List<SvgPathCommand> transformedCommands =
+      commands.map((SvgPathCommand c) => c.applyTransform(transform)).toList();
+    return new SvgPath(id, transformedCommands, opacity: opacity * transform.opacity);
+  }
+
+  @override
+  bool operator ==(Object other) {
+    if (runtimeType != other.runtimeType)
+      return false;
+    final SvgPath typedOther = other;
+    return id == typedOther.id
+        && opacity == typedOther.opacity
+        && const ListEquality<SvgPathCommand>().equals(commands, typedOther.commands);
+  }
+
+  @override
+  int get hashCode =>
+      id.hashCode ^
+      commands.hashCode ^
+      opacity.hashCode;
+
+  @override
+  String toString() {
+    return 'SvgPath(id: $id, opacity: $opacity, commands: $commands)';
+  }
+
+}
+
+/// Represents a single SVG path command from an SVG d element.
+///
+/// This class normalizes all the 'd' commands into a single type, that has
+/// a command type and a list of points.
+///
+/// Some examples of how d commands translated to SvgPathCommand:
+///   * "M 0.0, 1.0" => SvgPathCommand('M', [Point(0.0, 1.0)])
+///   * "Z" => SvgPathCommand('Z', [])
+///   * "C 1.0, 1.0 2.0, 2.0 3.0, 3.0" SvgPathCommand('C', [Point(1.0, 1.0),
+///      Point(2.0, 2.0), Point(3.0, 3.0)])
+class SvgPathCommand {
+  const SvgPathCommand(this.type, this.points);
+
+  /// The command type.
+  final String type;
+
+  /// List of points used by this command.
+  final List<Point<double>> points;
+
+  SvgPathCommand applyTransform(_Transform transform) {
+    final List<Point<double>> transformedPoints =
+    _vector3ArrayToPoints(
+        transform.transformMatrix.applyToVector3Array(
+            _pointsToVector3Array(points)
+        )
+    );
+    return new SvgPathCommand(type, transformedPoints);
+  }
+
+  @override
+  bool operator ==(Object other) {
+    if (runtimeType != other.runtimeType)
+      return false;
+    final SvgPathCommand typedOther = other;
+    return type == typedOther.type
+        && const ListEquality<Point<double>>().equals(points, typedOther.points);
+  }
+
+  @override
+  int get hashCode =>
+      type.hashCode ^
+      points.hashCode;
+
+  @override
+  String toString() {
+    return 'SvgPathCommand(type: $type, points: $points)';
+  }
+}
+
+class SvgPathCommandBuilder {
+  static const Map<String, Null> kRelativeCommands = const <String, Null> {
+    'c': null,
+    'l': null,
+    'm': null,
+    't': null,
+    's': null
+  };
+
+  Point<double> lastPoint = const Point<double>(0.0, 0.0);
+
+  SvgPathCommand build(String type, List<Point<double>> points) {
+    List<Point<double>> absPoints = points;
+    if (_isRelativeCommand(type)) {
+      absPoints = points.map((Point<double> p) => p + lastPoint).toList();
+    }
+
+    // closePath commands are always last in the path, so we don't need to track
+    // last point.
```

Comment:
they're last in a subpath, but are they always last in a path?

Reply:
e.g. isn't this a valid path? `M0,0 H1 V1 Z M2.2 H1 V1 Z`

Reply:
done